### PR TITLE
Multi file joli

### PR DESCRIPTION
**Pull request by mchenaux**

### CODE REVIEW

File: q2_snpsift/_snpsift.py [hunk 1]
```diff
@@ -25,16 +25,18 @@ def filter(
     filtered_vcf = VariantCallDir()
 
     with resources.path(bin, "SnpSift.jar") as executable_path:
-        cmd = [
+        for path, _ in input_vcf.vcf.iter_views(view_type=VariantCallFile):
+            cmd = [
             "java",
             "-jar",
-            executable_path,
+            executable_path,
```
Comment (HenziKou) [inline q2_snpsift/_snpsift.py:32]: Remove trailing whitespace.

File: q2_snpsift/_transformer.py [hunk 2]
```diff
@@ -23,11 +24,11 @@ def extract_fields_from_vcf(vcf_file: VariantCallDir) -> str:
...
     with resources.path(bin, "SnpSift.jar") as executable_path:
         cmd = [
-            "java",
-            "-jar",
-            executable_path,
-            "extractFields",
-            os.path.abspath(os.path.join(str(vcf_file), "vcf.vcf")),
+        "java",
+        "-jar",
+        executable_path,
```
Comment (HenziKou) [inline q2_snpsift/_transformer.py:29]: Remove trailing whitespace

File: q2_snpsift/_snpsift.py (hunk 1 above)
Comment (HenziKou) [inline q2_snpsift/_snpsift.py:32]: Indent everything inside the `cmd` brackets by one tab level.

File: q2_snpsift/_transformer.py (hunk 2 above)
Comment (HenziKou) [inline q2_snpsift/_transformer.py:29]: Indent everything inside the `cmd` brackets by one tab level.

File: q2_snpsift/_snpsift.py [hunk 3]
```diff
@@ -3,13 +3,13 @@
 import subprocess
 from importlib import resources
 
-from q2_types_variant import VariantCallAnnotationDir, VariantCallDir
-
 from q2_snpsift import bin
+from q2_types_variant import (VariantCallAnnotationDir, VariantCallDir,
+                              VariantCallFile, VCFIndexDirectory, VCFIndexFile)
```
Comment (HenziKou) [inline q2_snpsift/_snpsift.py:8]: Remove unused import `VCFIndexFile`

File: q2_snpsift/plugin_setup.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
 
 import importlib
 
+import q2_snpsift
 import qiime2.plugin
 from q2_types.feature_data import FeatureData
-from q2_types_variant import VariantCall, VariantCallAnnotation, Variants
+from q2_types_variant import (VariantCall, VariantCallAnnotation,
+                              VariantCallFile, Variants)
```
Comment (HenziKou) [inline q2_snpsift/plugin_setup.py:9]: Remove unused import `VariantCallFile`

**Review by HenziKou — CHANGES_REQUESTED** [review]

**Review by HenziKou — CHANGES_REQUESTED** [review]

File: q2_snpsift/_transformer.py
[the remark this answers is not in the document]
Comment (mchenaux) [reply q2_snpsift/_transformer.py:1]: Ok, so in the past couple of days I ran through the entire pipeline again in 2022.11 and got the whole thing working. There was a problem with the config file, which I regenerated. The 2022.11 qzas can be found here https://novozymes.sharepoint.com/:f:/r/sites/zynergy_3000031/Shared%20Documents/Variant_Calling_Pipeline/test_data?csf=1&web=1&e=XmdCFt including the config file that works.

[the remark this answers is not in the document]
Comment (mchenaux) [reply q2_snpsift/_transformer.py:1]: Ok I think i figured this out. I made a change in q2-types-variant on branch multi-sample that fixes this. It allows multiple files and names in the output. So, do we wait for that to be merged to main?

**Pull request merged**